src/github-api/implementation.ts

```diff
@@ -89,5 +98,36 @@ export function buildGitHubApi(token: string): GitHubApi {
         })
       );
     },
+    loadPullRequestStatus(pr) {
+      const query = gql`
+        query {
+          repository(owner: "${pr.repo.owner}", name: "${pr.repo.name}") {
+            pullRequest(number: ${pr.number}) {
+              reviewDecision
+              commits(last: 1) {
+                nodes {
+                  commit {
+                    statusCheckRollup {
+                      state
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      `;
+
+      return graphQLClient.request(query).then((response) => {
+        const result = response.repository.pullRequest;
+        const reviewDecision = result.reviewDecision;
+        const checkStatus =
+          result.commits.nodes[0].commit.statusCheckRollup?.state;
```

Comment:
Can `nodes` ever be empty? For example if you rebase a branch to match `main/master` exactly?

Perhaps worth adding a few extra `?` just in case:
```suggestion
          result.commits.nodes[0]?.commit?.statusCheckRollup?.state;
```

Reply:
added only to `nodes?.[0]?` to allow for no array or undef elem. `commit` should exist if there's a node